metrics-operator/converter/slo_converter.go

```diff
@@ -216,27 +243,58 @@ func cleanupCriteria(criteria []Criteria) []Criteria {
 	return newCriteria
 }
 
-// create operator for Target
-func newOperator(op string) (*metricsapi.Operator, error) {
-	// remove whitespaces
-	op = strings.Replace(op, " ", "", -1)
-
+// check if operator is valid and split it to operator and value
+func decodeOperatorAndValue(op string) (string, string, error) {
 	operators := []string{"<=", "<", ">=", ">"}
 	for _, operator := range operators {
 		if strings.HasPrefix(op, operator) {
-			return createOperator(operator, strings.TrimPrefix(op, operator))
+			return operator, strings.TrimPrefix(op, operator), nil
+		}
+	}
+
+	return "", "", fmt.Errorf(InvalidOperatorErrMsg, op)
+}
+
+// create operator for Target
+func newOperator(op []string, negate bool) (*metricsapi.Operator, error) {
+	// convert single operator
+	if len(op) == 1 {
+		operator, value, err := decodeOperatorAndValue(op[0])
+		if err != nil {
+			return nil, err
+		}
+		if negate {
+			return negateSingleOperator(operator, value)
+		} else {
+			return createSingleOperator(operator, value)
+		}
+	} else if len(op) >= 2 { // convert operators representing range
+		operator1, value1, err := decodeOperatorAndValue(op[0])
+		if err != nil {
+			return nil, err
+		}
+		operator2, value2, err := decodeOperatorAndValue(op[1])
+		if err != nil {
+			return nil, err
+		}
+		if negate {
+			return negateDoubleOperator(operator1, value1, operator2, value2)
+		} else {
+			return createDoubleOperator(operator1, value1, operator2, value2)
 		}
 	}
 
-	return &metricsapi.Operator{}, fmt.Errorf("invalid operator: '%s'", op)
+	return &metricsapi.Operator{}, fmt.Errorf("empty operators: '%v'", op)
 }
 
-// checks and negates the existing operator
-func createOperator(op string, value string) (*metricsapi.Operator, error) {
+// checks and negates the existing single operator
+//
+//nolint:dupl
+func negateSingleOperator(op string, value string) (*metricsapi.Operator, error) {
 	dec := inf.NewDec(1, 0)
 	_, ok := dec.SetString(value)
 	if !ok {
-		return nil, fmt.Errorf("unable to convert value '%s' to decimal", value)
+		return nil, fmt.Errorf(UnableConvertValueErrMsg, value)
```

Comment:
instead of having a string constant with a format specifier (%s), can we use a function that creates that error? The same goes for the `InvalidOperatorErrMsg`

Reply:
adapted

Reply:
did you commit that change? It seems like there are still a lot of occurences where the format strings are being used as constants

Reply:
yes, it should be commited, please try to refresh